features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -57,188 +272,1561 @@ const useStyles = makeStyles((theme) => ({
  */
 export default function Environments() {
     const classes = useStyles();
+    const intl = useIntl();
     const { api, updateAPI } = useContext(APIContext);
     const { settings } = useAppContext();
-    const [gatewayEnvironments, setGatewayEnvironments] = useState([...api.gatewayEnvironments]);
+    let revisionCount;
```

Comment:
This could have initialized as a const with the default value from the config as below

```js
const revisionCount = Configurations.app.revisionCount || 5;
```